### PR TITLE
Passing Early stopping params to SMC container for finetuning

### PR DESCRIPTION
**Pull request by kumar-shivam-ranjan**
This PR intends to add changes in create FT api to pass early stopping params to SMC container if provided by user
Screenshot 2024-10-17 at 7 33 11 PM

## UTs
Screenshot 2024-10-17 at 8 17 20 PM
Screenshot 2024-10-18 at 2 10 19 PM

### CODE REVIEW

File: ads/aqua/finetuning/finetuning.py [hunk 1]
```diff
@@ -522,9 +528,11 @@ def _build_oci_launch_cmd(
         val_set_size: float,
         parameters: AquaFineTuningParams,
         finetuning_params: str = None,
+        early_stopping_patience: int = None,
+        early_stopping_threshold: float = 0.0
     ) -> str:
         """Builds the oci launch cmd for fine tuning container runtime."""
-        oci_launch_cmd = f"--training_data {dataset_path} --output_dir {report_path} --val_set_size {val_set_size} "
+        oci_launch_cmd = f"--training_data {dataset_path} --output_dir {report_path} --val_set_size {val_set_size} --early_stopping_patience {early_stopping_patience} --early_stopping_threshold {early_stopping_threshold}"
```
Comment (qiuosier) [inline ads/aqua/finetuning/finetuning.py:535]: We should add the `early_stopping_patience` and `early_stopping_threshold` options only if `early_stopping_patience is not None`.

Comment (kumar-shivam-ranjan) [reply ads/aqua/finetuning/finetuning.py]: Updated. Thanks!

**Review by VipulMascarenhas — COMMENTED** [review]
changes look good, but unit tests will fail. Can you please fix them?

**Comment by kumar-shivam-ranjan** [comment]
> changes look good, but unit tests will fail. Can you please fix them?

Sure , will fix

**Comment by kumar-shivam-ranjan** [comment]
> changes look good, but unit tests will fail. Can you please fix them?

Done . Thanks

**Comment by qiuosier** [comment]
Just noticed that there is `AquaFineTuningParams` to contain all other parameters, maybe we should move the early stopping parameters into `AquaFineTuningParams`?

**Comment by kumar-shivam-ranjan** [comment]
> Just noticed that there is `AquaFineTuningParams` to contain all other parameters, maybe we should move the early stopping parameters into `AquaFineTuningParams`?

Done

**Review by qiuosier — APPROVED** [review]

**Pull request merged**